domain/src/main/java/com/anytypeio/anytype/domain/page/CreateObjectByTypeAndTemplate.kt

```diff
@@ -0,0 +1,130 @@
+package com.anytypeio.anytype.domain.page
+
+import com.anytypeio.anytype.core_models.Command
+import com.anytypeio.anytype.core_models.DVFilter
+import com.anytypeio.anytype.core_models.DVFilterCondition
+import com.anytypeio.anytype.core_models.DVSort
+import com.anytypeio.anytype.core_models.Id
+import com.anytypeio.anytype.core_models.InternalFlags
+import com.anytypeio.anytype.core_models.ObjectWrapper
+import com.anytypeio.anytype.core_models.Payload
+import com.anytypeio.anytype.core_models.Relations
+import com.anytypeio.anytype.core_models.Struct
+import com.anytypeio.anytype.core_models.primitives.SpaceId
+import com.anytypeio.anytype.core_models.primitives.TypeKey
+import com.anytypeio.anytype.domain.base.AppCoroutineDispatchers
+import com.anytypeio.anytype.domain.base.ResultInteractor
+import com.anytypeio.anytype.domain.block.repo.BlockRepository
+import com.anytypeio.anytype.domain.debugging.Logger
+import javax.inject.Inject
+
+class CreateObjectByTypeAndTemplate @Inject constructor(
+    private val repo: BlockRepository,
+    private val logger: Logger,
+    dispatchers: AppCoroutineDispatchers
+) : ResultInteractor<CreateObjectByTypeAndTemplate.Param,
+        CreateObjectByTypeAndTemplate.Result>(dispatchers.io) {
+
+    override suspend fun doWork(params: Param): Result {
+        val wrapper = searchObjectType(params)
+        logger.logInfo("CreateObjectByTypeAndTemplate, search object type:[$wrapper]")
+        return wrapper?.let {
+            createObject(
+                typeKey = params.typeKey,
+                template = it.defaultTemplateId,
+                internalFlags = params.internalFlags,
+                prefilled = params.prefilled,
+                space = params.space
+            )
+        } ?: throw RuntimeException("Object type not found")
```

Comment:
i guess it's better to wrap it in sealed class than throwing an exception. not being able to find an object doesn't mean an error, it might be an issue of syncing.

Reply:
fixed